client/src/pages/_document.tsx

```diff
@@ -1,13 +1,35 @@
-import { Html, Head, Main, NextScript } from 'next/document'
+import { ServerStyleSheets } from "@material-ui/core";
+import Document, { Head, Html, NextScript, Main } from "next/document";
+import React from "react";
 
-export default function Document() {
-  return (
-    <Html lang="en">
-      <Head />
-      <body>
-        <Main />
-        <NextScript />
-      </body>
-    </Html>
-  )
+export default class MyDocument extends Document {
+  render() {
+    return (
+      <Html lang="en">
+        <Head></Head>
+        <body>
+          <Main></Main>
+          <NextScript></NextScript>
+        </body>
+      </Html>
+    );
+  }
 }
```

Comment:
I'm still experimenting with the SSR issue and the `classNames` error, but this seems to be at least a stop-gap and actually renders the styling on initial render.